index.js

```diff
@@ -187,7 +187,7 @@ Choo.prototype.toString = function (location, state) {
   assert.equal(typeof location, 'string', 'choo.toString: location should be type string')
   assert.equal(typeof this.state, 'object', 'choo.toString: state should be type object')
 
-  this.state.href = location.replace(/\?*.$/, '')
+  this.state.href = location.replace(/\?.+$/, '')
```

Comment:
![image](https://user-images.githubusercontent.com/637410/30297738-6eaf365e-970e-11e7-9e58-323723469d17.png)

I believe `/\?.*?$/` is the winner, as the `+` operator modifier will not remove the "?" if that is all that's there.